pf4j/src/main/java/org/pf4j/PluginClassLoader.java

```diff
@@ -251,38 +224,37 @@ public URL getResource(String name) {
         return null;
     }
 
-    private URL findResourceFromDependencies(String name) {
+    protected URL findResourceFromDependencies(String name) {
         log.trace("Search in dependencies for resource '{}'", name);
         List<PluginDependency> dependencies = pluginDescriptor.getDependencies();
         for (PluginDependency dependency : dependencies) {
             PluginClassLoader classLoader = (PluginClassLoader) pluginManager.getPluginClassLoader(dependency.getPluginId());
 
             // If the dependency is marked as optional, its class loader might not be available.
-            if (classLoader == null && dependency.isOptional()) {
+            if (classLoader == null || dependency.isOptional()) {
```

Comment:
I think that is not good. You revert the bug solved by #386.

Reply:
oops solved